src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,63 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+
+    static final String DASH = " - ";
+    static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        // Initialize a StringBuilder to store the report
+        StringBuilder stringBuilder = new StringBuilder();
+        // Parse the dateFrom and dateTo strings into LocalDate objects
+        LocalDate fromDate = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate toDate = LocalDate.parse(dateTo, FORMATTER);
+
+        // Initialize an array to hold the LocalDate objects for each day in the data array
+        LocalDate[] dataArr = new LocalDate[data.length];
+
+        // Convert the dates in the data array into LocalDate objects
+        // and store them in the dataArr array
+        for (int i = 0; i < data.length; i++) {
+            dataArr[i] = LocalDate.parse(data[i].substring(0, 10), FORMATTER);
```

Comment:
Magic `0` and `10` should be constant. Don't use `substring`, try `split` instead